tools/ami-cleanup/src/internal/cleanup.go

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2024 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package internal
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/account"
+	accountTypes "github.com/aws/aws-sdk-go-v2/service/account/types"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/gravitational/trace"
+	"github.com/relvacode/iso8601"
+)
+
+type ApplicationInstance struct {
+	shouldDoDryRun bool
+
+	// Dependency injection for testing
+	accountClientGenerator func(cfg *aws.Config) IAccountApi
+	ec2ClientGenerator     func(cfg *aws.Config) IEc2Api
+}
+
+// Creates a new instance of the tool
```

Comment:
Let's follow the style guide for godocs: https://go.dev/wiki/CodeReviewComments#comment-sentences.
```suggestion
// NewApplicationInstance creates a new instance of the tool.
```